src/main/java/com/example/demo/config/SecurityConfig.java

```diff
@@ -0,0 +1,42 @@
+package com.example.demo.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.stereotype.Component;
+
+@EnableMethodSecurity
+@RequiredArgsConstructor
+@Component
```

Comment:
Please investigate difference between `@Component` and `@Configuration`. We use `@Bean` only with `@Configuration`, otherwise it may cause bugs in your program.
```suggestion
@Configuration
```